component/factories.go

```diff
@@ -32,6 +32,9 @@ type Factories struct {
 
 	// Extensions maps extension type names in the config to the respective factory.
 	Extensions map[Type]ExtensionFactory
+
+	// Connectors maps connector type names in the config to the respective factory.
+	Connectors map[Type]ConnectorFactory
```

Comment:
Since the factory was added to `connectors` package in #6611, this would ideally be changed:
```suggestion
	Connectors map[Type]connector.Factory
```

However, this causes a circular dependency.

@bogdandrutu, is the plan still to move `Factories` to the `service` package, similar to #6552?

Reply:
Yes :) You can create a PR for that, would be great.